### PR TITLE
AWS Rightsize EC2 Instances Revamp

### PR DESCRIPTION
**Pull request by XOmniverse**
### Description

This is a full revamp of the AWS Rightsize EC2 Instances. All functionality has been tested, but I encourage others to double check my work.

From the CHANGELOG:

- Policy name changed to reference EC2 service directly
- Several parameters altered to be more descriptive and human-readable
- Removed deprecated "Log to CM Audit Entries" parameter
- Added ability to only report recommendations that meet a minimum savings threshold
- Added ability to configure how many days to consider CPU/memory statistics for
- Added ability to filter resources by multiple tag key:value pairs
- Added ability to take automated actions to resize or delete resources
- Added ability to make recommendations based on maximum CPU/memory usage
- Added additional context to incident description
- Normalized incident export to be consistent with other policies
- Added human-readable recommendation to incident export
- Added additional fields to incident export
- Policy no longer raises new escalations if statistics or savings data changed but nothing else has
- Streamlined code for better readability and faster execution

### Link to Example Applied Policy

https://app.flexera.com/orgs/6/automation/applied-policies/projects/7954?policyId=64b97fec1c05c900018b2b34

### Contribution Check List

- [X] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable
- [X] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

**Comment by XOmniverse** [comment]
> @XOmniverse - think you also need to update top level README with the new policy name
> 
> Also, should the path to policy change? e.g., from `cost/aws/rightsize_compute_instances/aws_compute_rightsizing.pt` to `cost/aws/rightsize_ec2_instances/aws_ec2_rightsizing.pt`

For the first item, the global README is now updated through automation.

For the second item, I agree and have made this change.

**Comment by nia-vf1** [comment]
Not sure why it is complaining about not having a README, could be because we moved it? Anyway, PR looks fine

File: tools/cloudformation-template/FlexeraAutomationPolicies.template [hunk 1]
```diff
@@ -404,6 +422,20 @@ Mappings:
...
+        - "ec2:DescribeTags"
+        - "cloudwatch:GetMetricStatistics"
+        - "cloudwatch:GetMetricData"
+        - "cloudwatch:ListMetrics"
+      action:
+        - "ec2:DescribeInstanceStatus"
+        - "ec2:ModifyInstanceAttribute"
+        - "ec2:StartInstances"
+        - "ec2:StopInstances"
+        - "ec2:TerminateInstances"
```
Comment (bryankaraffa) [inline tools/cloudformation-template/FlexeraAutomationPolicies.template:438]:
```suggestion
    AWSRightsizeEC2Instances:
      read:
        - "ec2:DescribeInstanceStatus"
        - "ec2:DescribeRegions"
        - "ec2:DescribeInstances"
        - "ec2:DescribeTags"
        - "cloudwatch:GetMetricStatistics"
        - "cloudwatch:GetMetricData"
        - "cloudwatch:ListMetrics"
      action:
        - "ec2:ModifyInstanceAttribute"
        - "ec2:StartInstances"
        - "ec2:StopInstances"
        - "ec2:TerminateInstances"
```
Same as https://github.com/flexera-public/policy_templates/pull/1308/files#r1273620627

**Review by bryankaraffa — DISMISSED** [review]

**Review by taterdog590 — APPROVED** [review]
STAMP

**Review by nia-vf1 — APPROVED** [review]
LGTM

**Pull request merged**